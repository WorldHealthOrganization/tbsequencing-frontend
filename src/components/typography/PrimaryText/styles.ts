import { css } from '@emotion/react';
import { font, fontSizes } from '../fontSizes';

export const primaryTextStyles = css({
  fontFamily: font.regular,
  fontSize: fontSizes.primary,
  lineHeight: '20px',
});
