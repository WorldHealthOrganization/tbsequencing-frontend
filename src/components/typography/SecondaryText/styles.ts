import { css } from '@emotion/react';
import { font, fontSizes } from '../fontSizes';

export const secondaryTextStyles = css({
  fontSize: fontSizes.secondary,
  fontFamily: font.regular,
  lineHeight: '20px',
});
