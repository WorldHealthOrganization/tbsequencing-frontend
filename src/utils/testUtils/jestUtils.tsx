import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
