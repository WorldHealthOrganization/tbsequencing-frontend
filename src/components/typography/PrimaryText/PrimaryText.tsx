/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import BaseText from '../BaseText';

type InjectedStyle = SerializedStyles | Array<SerializedStyles | undefined>;

interface Props {
  children: ReactNode;
  style?: InjectedStyle;
}

export const getInjectedStyles = (
  injectedStyle: InjectedStyle | undefined,
  baseStyle: SerializedStyles,
) => {
  if (!injectedStyle) {
    return [baseStyle];
  }

  if (Array.isArray(injectedStyle)) {
    return [baseStyle, ...injectedStyle];
  }

  return [baseStyle, injectedStyle];
};

export const PrimaryText = ({ children, style }: Props) => (
  <BaseText
    style={getInjectedStyles(style, styles.primaryTextStyles)}
  >
    {children}
  </BaseText>
);

export default PrimaryText;
