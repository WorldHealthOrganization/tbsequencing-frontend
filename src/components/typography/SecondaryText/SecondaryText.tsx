/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { css, SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import BaseText from '../BaseText';

interface Props {
  children: ReactNode;
  style?: SerializedStyles;
}

export const SecondaryText = ({ children, style }: Props) => {
  const injectedStyles = style || css({});

  return (
    <BaseText
      style={[styles.secondaryTextStyles, injectedStyles]}
    >
      {children}
    </BaseText>
  );
};

export default SecondaryText;
