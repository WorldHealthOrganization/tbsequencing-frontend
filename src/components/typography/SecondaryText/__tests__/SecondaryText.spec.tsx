import React from 'react';
import { getTreeJSONWithRedux, render, screen } from '../../../../utils/testUtils/jestUtils';
import SecondaryText from '../SecondaryText';
import PrimaryText from '../../PrimaryText';

describe('SecondaryText component test', () => {
  const mockText = 'mockText';

  test('renders SecondaryText component with a proper text', () => {
    render(
      <SecondaryText>{mockText}</SecondaryText>,
    );

    const foundElement = screen.queryByText(mockText);

    expect(foundElement).toBeTruthy();
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <PrimaryText>{mockText}</PrimaryText>,
    );

    expect(tree).toMatchSnapshot();
  });
});
