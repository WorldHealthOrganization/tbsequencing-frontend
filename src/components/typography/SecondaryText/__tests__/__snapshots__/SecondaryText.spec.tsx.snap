// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryText component test Should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
>
  mockText
</p>
`;
